### DISCUSSION TITLE
Allow config flow for calculated integrations without dependencies to other integrations/entities

### DISCUSSION THREAD

**Discussion by frenck** [home-assistant/architecture#569] [General]
## Context

Currently, we allow config flows for integrations that connect to devices or services. This is the bigger amount of integrations Home Assistant provides and steadily growing.

Besides the configuration flow, we have a UI for helpers, like all the `input_*` platforms.

However, this means, we still have gaps:

- Integrations that consume and process data from other integrations; ex: Utility meter
- Integrations that calculate their state locally, entirely by itself, without relying on other integrations; ex: Sun, Moon, Season.

## Proposal

I'd like to propose that we allow the use of configuration flows for the following case:

> Integrations that calculate their state locally, entirely by itself, without relying on other integrations; ex: Sun, Moon, Season.

These are used a lot by a lot of installations but are currently not available via the UI, but used quite a lot:

- Sun: 99,1%
- Workday: 6,4%
- Uptime: 5.6%
- Moon: 3,4%
- Season: 2,3%

Sun is a weird one because we enable it by default on every installation. But things like workday and season are really helpful in automations; and they are only usable if you are using YAML 🤷 

## Consequences

... We've got some work to do?

**Comment by frenck** [comment]
The only thing that I see we should think about, is the service type.
We currently define either `services` or `devices`, this would bring in...?

**Comment by balloob** [comment]
I am not opposed, but I wonder if there are other alternatives.

One alternative I can think of is having these just be a list of toggles that one can enable. 

- Maintain a sun entity
- Maintain a season entity
- Maintain a moon entity

Unless we're afraid that we will get thousands of these.

**Reply by frenck** [reply]
Well, we have quite a few of these already.

I think the main consideration are things like workdays, holidays and those. One could one more of them, additionally, they could provide settings/configuration as well. I agree a checkbox for "sun" would be nice, however, for workday or holiday, things become different really quick.

Considering that, I think adding just an integration, would be re-using all things we have already (config flow, option flow, rules/testing) and limits the places a user has to go.

I think from that perspective, allowing config flow makes sense. It can be done now, everybody is used to it, and bring instant usability for the end-user.

**Reply by dgomes** [reply]
I've been looking for an opportunity to add a UI and/or discovery to the utility_meter integration, which is way more then a checkbox.

**Reply by frenck** [reply]
The utility_meter integration is not included in my proposal, and would not be allowed for this in this proposal, as it consumes data from other integrations.

**Reply by javicalle** [reply]
From an user point of view, I don't have to know if an integration connects to a service or not, so . For that reason, it is not trivial to the average user to determine whether the integration should be configured in the standard way or not. So IMHO it will be better use the same configuration mechanism.

In addition, it could be that some integration evolves and requires connecting to some type of service. I am thinking of some resolutions for the holidays with the Hebrew or Muslim calendars (can't remember which one), which cannot be calculated. If in a future, the integration will require connect to a service, then the configuration should be migrated to the 'other' type of integrations.
